docs/page-objects/testing.rst

```diff
@@ -322,3 +322,42 @@ The coverage for page object code is reported correctly if tools such as
 `coverage`_ are used when running web-poet tests.
 
 .. _coverage: https://coverage.readthedocs.io/
+
+Item adapters
+=============
+
+The testing framework uses the itemadapter_ library to convert items to dicts
+when storing them in fixtures and when comparing the expected and the actual
+output. As adapters may influence the resulting dicts, it's important to use
+the same adapter in both cases and it may be useful to use a different adapter
+in tests and in production (for example, you may want to omit empty fields in
+production but be able to distinguish between empty and absent fields in
+tests). For this you can set the ``adapter_type_name`` field in the metadata
+dictionary to the type name of a class that inherits from
+:class:`itemadapter.ItemAdapter` and has the adapter(s) you want to use in
+tests in its ``ADAPTER_CLASSES`` attribute (see `the relevant itemadapter
+docs`_ for more information). An example::
+
+    from collections import deque
+
+    from itemadapter import ItemAdapter
+    from itemadapter.adapter import DictAdapter
+
+
+    class MyAdapter(DictAdapter):
+        # any needed customization
+        ...
+
+    class MyItemAdapter(ItemAdapter):
+        ADAPTER_CLASSES = deque([MyAdapter])
+
+You can then put the fully qualified name of ``MyItemAdapter`` into
+``adapter_type_name`` and it will be used by the testing framework.
```

Comment:
Are we using "type name" elsewhere? For MyItemAdapter I'd expect type name to be MyItemAdapter. In Scrapy docs we use "import path" name, see https://docs.scrapy.org/en/latest/topics/settings.html#import-paths-and-classes. I guess this applies both to docs and to the meta key names.

Reply:
The serialized exception has "type_name", though it's less user-visible. `adapter_import_path` sounds fine.

Reply:
we might change type_name there as well; backwards compat should be easy to achieve, and it's maybe not even necessary - release a new version, move on :)

Reply:
Would it make sense to call it `adapter` and expect the class object itself? I imagine scrapy-poet allowing to set this with an import path string in a setting, but scrapy-poet could load the class before passing it to web-poet.